raiden/network/proxies/token_network.py

```diff
@@ -809,96 +861,178 @@ def _set_total_deposit(
                 partner=partner,
             )
             self.client.poll(transaction_hash)
-            receipt_or_none = check_transaction_threw(self.client, transaction_hash)
+            receipt = check_transaction_threw(self.client, transaction_hash)
 
-        transaction_executed = gas_limit is not None
-        if not transaction_executed or receipt_or_none:
-            if transaction_executed:
-                block = receipt_or_none["blockNumber"]
-            else:
-                block = checking_block
+            if receipt:
```

Comment:
The following checks are missing:

- safety_deprecation_switch
- token_network_deposit_limit
- channel_participant_deposit_limit

The difference is that on preconditions they are preconditions violations (BrokenPreconditionError). Here there are race conditions. (RecoverableErrors)

Reply:
Those 3 checks were added and raise `RaidenRecoverableError`